src/scikit_build_cli/utils.py

```diff
@@ -20,25 +19,15 @@ def __dir__() -> list[str]:
     return __all__
 
 
-def _build_dir(func: F) -> F:
```

Comment:
About the decorator approach, it can be more flexible because we can add multiple options to one decorator, even group them together so that they are merged into one kwargs
https://github.com/LecrisUT/fmf/blob/b9495adabe37d52a0db46bacb685d8542a927a45/fmf/cli.py#L49-L77
It's just unfortunate that the mypy is so hard to figure around it

Reply:
If you do need to do it, then you can use [ParamSpec](https://docs.python.org/3/library/typing.html#typing.ParamSpec). It's a little weird and limited, but this is what it's good at. Though I'd only introduce that if necessary.

For simplicity, this is the Python 3.12+ only syntax:

```python
def _build_dir[T, **P](f: Callable[P, T]) -> Callable[P, T]:
   """Add build_dir click option"""

    @click.option(
        "--build-dir",
        "-B",
        type=click.Path(
            exists=True,
            file_okay=False,
            dir_okay=True,
            writable=True,
            path_type=pathlib.Path,
        ),
        help="Path to cmake build directory",
    )
    @functools.wraps(func)
    def wrapper(*args: P.args, **kwargs: P.kwargs):
        return func(*args, **kwargs)

    return wrapper
```

(Without 3.12, you'd need 

```python
T = TypeVar('T')
P = ParamSpec('P')
```
)

Reply:
I've tried it at some point, but it didn't work for me. I need to try it again. The main issues were casting of wrapper, needing to concatanate the `build_dir` argument, and maybe something about functools. Do you have an example in the wild where click was configured with those?

Reply:
I don't use click all that heavily. This is the only place I can think of quickly: https://github.com/henryiii/hypernewsviewer/blob/c239bbaa8e795dcffb119ca20df0c30a8dda432a/src/hypernewsviewer/model/__main__.py#L76-L88